src/components/Publication/FullPublication.tsx

```diff
@@ -3,24 +3,30 @@ import type { LensterPublication } from '@generated/lenstertypes';
 import getAppName from '@lib/getAppName';
 import dayjs from 'dayjs';
 import relativeTime from 'dayjs/plugin/relativeTime';
-import type { FC } from 'react';
+import type { FC, Ref } from 'react';
+import { useRef } from 'react';
+import { useEffect } from 'react';
 
 import PublicationActions from './Actions';
 import HiddenPublication from './HiddenPublication';
 import PublicationBody from './PublicationBody';
 import PublicationStats from './PublicationStats';
-import PublicationType from './Type';
+import ThreadBody from './ThreadBody';
 
 dayjs.extend(relativeTime);
 
 interface Props {
   publication: LensterPublication;
+  postContainerRef?: Ref<HTMLDivElement>;
 }
 
-const FullPublication: FC<Props> = ({ publication }) => {
+const FullPublication: FC<Props> = ({ publication, postContainerRef }) => {
+  const threadRef = useRef<HTMLDivElement>(null);
   const isMirror = publication.__typename === 'Mirror';
   const profile = isMirror ? publication?.mirrorOf?.profile : publication?.profile;
   const timestamp = isMirror ? publication?.mirrorOf?.createdAt : publication?.createdAt;
+  const commentOn = publication.commentOn as LensterPublication;
```

Comment:
I have noticed that optional chaining is being used for `publication` in few other places. Can we use it here and some other missing places too, to prevent breaking the app accidentally by any chance?
```suggestion
  const isMirror = publication?.__typename === 'Mirror';
  const profile = isMirror ? publication?.mirrorOf?.profile : publication?.profile;
  const timestamp = isMirror ? publication?.mirrorOf?.createdAt : publication?.createdAt;
  const commentOn = publication?.commentOn as LensterPublication;
```